app/models/thread.php

```diff
@@ -83,38 +80,131 @@ public static function get($id)
         return new self($row);
     }
 
-     public function getComments($offset, $limit)
-    {
-        $comments = array();
-        $db = DB::conn();
-        $rows = $db->rows("SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($this->id));
-        foreach($rows as $row) {
-        $comments[] = new Comment($row);
-        }
-        return $comments;
-    }
-
     public function write(Comment $comment)
     {
         if(!$comment->validate()) {
             throw new ValidationException('invalid comment');
         }
 
-        $db = DB::conn();
-        
         try {
+            $db = DB::conn();
+            $date_created = date("Y-m-d H:i:s");
+
             $db->begin();
-            $db->insert(
-                'comment', array(
-                    'thread_id' => $this->id, 
-                    'username' => $comment->username, 
-                    'body' => $comment->body
-                    )
-                );
+            $params = array(
+                        'thread_id' => $this->id, 
+                        'user_id' => $_SESSION['user_id'],
+                        'body' => $comment->body,
+                        'created' => $date_created
+                        );
+            $db->insert('comment', $params);
             $db->commit();
-        
+                
         }catch (Exception $e) {
             $db->rollback();
         }
     }
+
+    public function isUserThread()
+    {
+        return $this->user_id === $_SESSION['user_id'];
+    }
+
+    public function delete()
+    {
+        try {
+            $db = DB::conn();
+
+            $db->begin();
+            $db->query('DELETE FROM thread WHERE id = ?', array($this->id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+
+    }
+
+    public function addFollow()
+    {
+        try {
+            $db = DB::conn();
+
+            $db->begin();
+            $params = array(
+                        'thread_id' => $this->id,
+                         'user_id' => $_SESSION['user_id'] 
+                         );
+
+            $db->insert('follow', $params);
+            $db->commit();
+        }   catch (Exception $e) {
```

Comment:
There should only be 1 space between the curly brace and before the catch statement po :bow: 
```
} catch (Exception $e) {
    ...
}
``` :bow: